tests/ui/min_rust_version_attr.rs

```diff
@@ -0,0 +1,51 @@
+#![allow(clippy::redundant_clone)]
+#![feature(custom_inner_attributes)]
+#![clippy::msrv = "1.0.0"]
```

Comment:
The standard library's APIs are never added between patch versions, so I don't think it needs to enter the patch version. (Always assign 0 when comparing)
This (omit patch version) is also consistent with the version format used in [`cfg(version(..))`](https://github.com/rust-lang/rfcs/blob/master/text/2523-cfg-path-version.md#summary) and [MSRV RFC](https://github.com/rust-lang/rfcs/blob/master/text/2495-min-rust-version.md).

Reply:
Perhaps `semver::Version::parse` already accepts it? In that case, it would be nice to have a test to make sure it's okay to omit the patch version.

Reply:
I expect, that this should Just Work. But a test would indeed be nice.

Reply:
@flip1995, @taiki-e `Version::parse` throws an error if the patch version is omitted ☹️ 
The docs for the `parse` also mention that it needs exactly 3 parameters 
https://docs.rs/semver/0.11.0/semver/struct.Version.html#method.parse

Reply:
`VersionReq` sounds good. We then also would support `=1.40.0` and similar. Not that it would be necessary, but nice to have.